pkg/client/events/replay_client.go

```diff
@@ -87,20 +89,19 @@ type replayClient[T any, U observable.ReplayObservable[T]] struct {
 func NewEventsReplayClient[T any, U observable.ReplayObservable[T]](
 	ctx context.Context,
 	deps depinject.Config,
-	cometWebsocketURL string,
 	queryString string,
 	newEventFn NewEventsFn[T],
 	replayObsBufferSize int,
 ) (client.EventsReplayClient[T, U], error) {
 	// Initialize the replay client
 	rClient := &replayClient[T, U]{
-		endpointURL:  cometWebsocketURL,
-		queryString:  queryString,
-		eventDecoder: newEventFn,
+		queryString:         queryString,
+		eventDecoder:        newEventFn,
+		replayObsBufferSize: replayObsBufferSize,
 	}
 	replayObsCache, replayObsCachePublishCh := channel.NewReplayObservable[U](
 		ctx,
-		replayObsBufferSize,
+		1, // replay buffer size 1
```

Comment:
NIT: maybe we should add a comment similar to that above as a note that this will go away.

Reply:
I feel like otherwise, I would like the comment to speak to the "why" a bit more (the cache only ever needs to return the last cached value).